x-pack/legacy/plugins/graph/public/app.js

```diff
@@ -59,20 +59,22 @@ export function initGraphApp(angularModule, deps) {
     Storage,
     canEditDrillDownUrls,
     graphSavePolicy,
+    licensing,
   } = deps;
 
   const app = angularModule;
 
-  function checkLicense(kbnBaseUrl) {
-    const licenseAllowsToShowThisPage = xpackInfo.get('features.graph.showAppLink') &&
-      xpackInfo.get('features.graph.enableAppLink');
+
+  licensing.license$.subscribe(license => {
+    const info = checkLicense(license);
```

Comment:
Just a note: in the new platform we run all applications, it means that application code can be executed on any page. I'm not familiar with your setup, probably it's not a problem now.

Reply:
Good point. For now Graph has its own route and will cause a full page reload on navigating to another app. I changed the implementation to correctly unsubscribe on application unmount though to not forget it when moving it over to the new platform.